lib/AST/Decl.cpp

```diff
@@ -6473,6 +6473,12 @@ static bool requiresNewVTableEntry(const AbstractFunctionDecl *decl) {
   auto baseInterfaceTy = base->getInterfaceType();
   auto derivedInterfaceTy = decl->getInterfaceType();
 
+  using Direction = ASTContext::OverrideGenericSignatureReqCheck;
+  if (!ctx.overrideGenericSignatureReqsSatisfied(
```

Comment:
I don’t think this cast is necessary.

Reply:
I added it because otherwise I get an error: `Cannot initialize a parameter of type 'swift::ValueDecl *' with an lvalue of type 'const swift::AbstractFunctionDecl *'`

Reply:
Ah, I see. The right thing would be to make the parameters to overrideGenericSignatureReqsSatisfied() `const`.

Reply:
@slavapestov Ok done :)